### PR TITLE
[10.x] Fixes `whereDate`, `whereDay`, `whereMonth`, `whereTime`, `whereYear` and `whereJsonLength` to ignore invalid `$operator`

### PR DESCRIPTION
**Pull request by crynobone**

### CODE REVIEW

File: src/Illuminate/Database/DBAL/TimestampType.php [hunk 1]
```diff
@@ -33,13 +36,16 @@ public function getSQLDeclaration(array $column, AbstractPlatform $platform): st
             MySQLPlatform::class,
             MySQL57Platform::class,
             MySQL80Platform::class,
+            MySQL84Platform::class,
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/DBAL/TimestampType.php:39]: Shouldn't we move these changes to a separate PR?

**Comment by rodrigopedra** [comment]
Mind if I ask why? Or what is this trying to solve?

Isn't it better to throw an `InvalidArgumentException`?

I mean, if a particular DBMS supports an operator not listed as valid within the class, I guess the developer would be surprised by this behavior.

File: src/Illuminate/Database/DBAL/TimestampType.php (hunk 1 above)
Comment (crynobone) [reply src/Illuminate/Database/DBAL/TimestampType.php:39]: 10.x is currently on security only updates.

**Comment by GrahamCampbell** [comment]
> Mind if I ask why? Or what is this trying to solve?

I assume the issue is SQL injection, where the operator comes from user input? We have a similar fallback system for when an invalid sort direction is passed to order by, for example.
